test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -247,7 +247,115 @@ Describe "Invoke-WebRequest tests" -Tags "Feature" {
 
         $result = ExecuteWebCommand -command $command
         $result.Error.FullyQualifiedErrorId | Should Be "WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand"
+    }
+
+    It "Validate Invoke-WebRequest returns User-Agent where -NoProxy with -Proxy set" {
```

Comment:
If we agree that `-NoProxy` and `-Proxy` are mutually exclusive, then this test should validate you get an `AmbiguousParameterSet,Microsoft.PowerShell.Commands.NewPSSessionCommand` exception